be/src/block_cache/block_cache.cpp

```diff
@@ -0,0 +1,135 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "block_cache/block_cache.h"
+
+#include <fmt/format.h>
+
+#include "block_cache/fb_cachelib.h"
+#include "common/config.h"
+#include "common/logging.h"
+#include "common/statusor.h"
+#include "gutil/strings/substitute.h"
+
+namespace starrocks {
+
+BlockCache::BlockCache() {
+    _kv_cache = std::make_unique<FbCacheLib>();
+}
+
+BlockCache::~BlockCache() {
+    if (_kv_cache) {
+        _kv_cache->destroy();
+    }
+}
+
+BlockCache* BlockCache::instance() {
+    static BlockCache cache;
+    return &cache;
+}
+
+Status BlockCache::init(const CacheOptions& options) {
+    return _kv_cache->init(options);
+}
+
+Status BlockCache::set_block_size(size_t block_size) {
```

Comment:
looks like block_size is a key parameter to the cache system, changing block_size will invalidate all cache's index. Whether is a good idea expose such dangerous  interface? or better make it part of CacheOption and immutable after init?

Reply:
> looks like block_size is a key parameter to the cache system, changing block_size will invalidate all cache's index. Whether is a good idea expose such dangerous interface? or better make it part of CacheOption and immutable after init?

A `TODO` comment was left above the interface, now that you mentioned, I will update it in this version.